src/core/socket.h

```diff
@@ -162,11 +167,18 @@ struct Socket {
         RateLimit trigger_limit;
 };
 
+typedef struct SocketPeer {
+        unsigned n_connections;
+
+        int family;
+        union sockaddr_union peer;
+} SocketPeer;
```

Comment:
Yes I changed it now . I was thinking about putting the peer pointer but then din't wanted to add a extra variable for Service structure.